tests/jobs/test_backfill_job.py

```diff
@@ -139,14 +167,14 @@ def test_unfinished_dag_runs_set_to_failed(self, dag_maker):
 
         assert State.FAILED == dag_run.state
 
-    def test_dag_run_with_finished_tasks_set_to_success(self, dag_maker):
+    def test_dag_run_with_finished_tasks_set_to_success(self, dag_maker, mock_executor):
```

Comment:
It feels weird to specify a fixture and not using it directly, just to using it for its mock. So maybe define an additional fixture called `mock_job`?

Reply:
Another common pattern is to use a marker instead, see `need_serialized_dag`.

Reply:
You can set fixture to be autoused for your set of tests: https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#autouse-fixtures-fixtures-you-don-t-have-to-request - that's the best pattern I think

Reply:
Many of the tests actually do use the object, which is why it is yielded from the fixture. And to me I think it is actually quite common to see a fixture being used without using the object it yields, you'll see this across the Airflow code base and many others.

I don't want to use auto-use since not all tests need or want the executor mocked in this way (autouse can be a source of great confusion when writing tests and you're not sure why the behaviour doesn't match your expectations).